### PR TITLE
Checkbox to display joints in radians

### PR DESCRIPTION
**Pull request by aaryanmurgunde**
### Description

Issue #3513 Add a checkbox to display joints in radians
Need help to connect it to the flag.

image
image

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [x] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

### CODE REVIEW

**Review by rhaschke — COMMENTED** [review]
I suggest using radio buttons (to see both options). I committed appropriate changes and (hopefully) linked the buttons to the delegate. I didn't test (or even compile) because I don't have the resources right now. Please continue :wink:

File: moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_joints_widget.cpp [hunk 1]
```diff
@@ -192,7 +192,12 @@ MotionPlanningFrameJointsWidget::MotionPlanningFrameJointsWidget(MotionPlanningD
   ui_->joints_view_->viewport()->installEventFilter(new JointsWidgetEventFilter(ui_->joints_view_));
   // intercept keyboard events delivered to joints_view_ to operate joints directly
   ui_->joints_view_->installEventFilter(new JointsWidgetEventFilter(ui_->joints_view_));
-  ui_->joints_view_->setItemDelegateForColumn(1, new ProgressBarDelegate(this));
+  auto delegate = new ProgressBarDelegate(this);
+  connect(ui_->joints_view_->button_group_units_, QButtonGroup::idToggled, [delegate](int id, bool checked) {
```
[the remark this answers is not in the document]
Comment (aaryanmurgunde) [reply moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_joints_widget.cpp:196]: But still it gives an error on connect saying it does not have the overloading for these types
image

image

**Review by aaryanmurgunde — COMMENTED** [review]
Thank you soo much for this 😃 unfortunately there is a minor connect issue while compiling. Tried a few things but they dont work

File: moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_joints_widget.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (aaryanmurgunde) [reply moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_joints_widget.cpp:196]: As suggested here : [StackOverflow](https://stackoverflow.com/questions/57507040/qt-buttongroup-and-checkable-pushbuttons-how-to-connect-toggled-signal-with-int)

image

[the remark this answers is not in the document]
Comment (aaryanmurgunde) [reply moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_joints_widget.cpp:196]: Found a solution [Qt 5.2 Update](https://doc.qt.io/archives/qt-5.6/qbuttongroup.html#buttonToggled-1)

```cpp
connect(buttonGroup, static_cast<void(QButtonGroup::*)(int, bool)>(&QButtonGroup::buttonToggled),
    [=](int id, bool checked){ /* ... */ });
```
```cpp
connect(ui_->button_group_units_, static_cast<void(QButtonGroup::*)(int)>(&QButtonGroup::buttonClicked), [delegate](int id, bool checked=true) {
    if (checked)
      delegate->setUnit(id == 0 ? ProgressBarDelegate::Degrees : ProgressBarDelegate::Radians);
  });

```

**Comment by aaryanmurgunde** [comment]
## Current issue : Trigger working when displaying from degrees to radians but once clicked to radians does not go back

**Review by rhaschke — APPROVED** [review]
I fixed the remaining issues and added saving/restoring to/from rviz config.

**Comment by aaryanmurgunde** [comment]
@rhaschke Thank you so much for your help ! 🤩

**Comment by v4hn** [comment]
Thank you so much for addressing that small but very annoying UI issue. :fireworks:

**Pull request merged**